clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractHeartbeatRequestManager.java

```diff
@@ -375,81 +342,88 @@ private void onErrorResponse(final ConsumerGroupHeartbeatResponse response,
-                logger.error("GroupHeartbeatRequest failed due to unreleased instance id {}: {}",
-                        membershipManager.groupInstanceId().orElse("null"), errorMessage);
-                handleFatalFailure(error.exception(errorMessage));
-                break;
-
-            case FENCED_INSTANCE_ID:
-                logger.error("GroupHeartbeatRequest failed due to fenced instance id {}: {}. " +
-                        "This is expected in the case that the member was removed from the group " +
-                        "by an admin client, and another member joined using the same group instance id.",
-                    membershipManager.groupInstanceId().orElse("null"), errorMessage);
-                handleFatalFailure(error.exception(errorMessage));
-                break;
-
             case INVALID_REQUEST:
             case GROUP_MAX_SIZE_REACHED:
             case UNSUPPORTED_ASSIGNOR:
             case UNSUPPORTED_VERSION:
-                logger.error("GroupHeartbeatRequest failed due to {}: {}", error, errorMessage);
+                logger.error("{} failed due to {}: {}", heartbeatRequestName(), error, errorMessage);
                 handleFatalFailure(error.exception(errorMessage));
                 break;
 
             case FENCED_MEMBER_EPOCH:
-                message = String.format("GroupHeartbeatRequest failed for member %s because epoch %s is fenced.",
-                        membershipManager.memberId(), membershipManager.memberEpoch());
+                message = String.format("%s failed for member %s because epoch %s is fenced.",
+                        heartbeatRequestName(), membershipManager().memberId(), membershipManager().memberEpoch());
                 logInfo(message, response, currentTimeMs);
-                membershipManager.transitionToFenced();
+                membershipManager().transitionToFenced();
                 // Skip backoff so that a next HB to rejoin is sent as soon as the fenced member releases its assignment
                 heartbeatRequestState.reset();
                 break;
 
             case UNKNOWN_MEMBER_ID:
-                message = String.format("GroupHeartbeatRequest failed because member %s is unknown.",
-                        membershipManager.memberId());
+                message = String.format("%s failed because member %s is unknown.",
+                        heartbeatRequestName(), membershipManager().memberId());
                 logInfo(message, response, currentTimeMs);
-                membershipManager.transitionToFenced();
+                membershipManager().transitionToFenced();
                 // Skip backoff so that a next HB to rejoin is sent as soon as the fenced member releases its assignment
                 heartbeatRequestState.reset();
                 break;
 
             default:
                 // If the manager receives an unknown error - there could be a bug in the code or a new error code
```

Comment:
should we move this comment inside the `if`? at this point we're not yet in the "unknown" errors land anymore